site-src/geps/gep-917.md

```diff
@@ -126,23 +124,46 @@ Fields marked "Extended" support will eventually have conformance tests that
 lock in the behavior of that feature, and there will be a mechanism for implementations
 to tell the testing framework what extended fields they support.
 
-#### Testing framework
+## Testing framework
+
+Conformance tests will initially be structured as a CLI that runs on existing
+Kubernetes clusters. This may be expanded to include a testing library that can
+be imported by implementations. This tool will output test results in a
+structured format that can be used to generate reports. Where possible it will
+share code with the existing ingress-controller-conformance test suite.
```

Comment:
For your considerations: while I think a CLI is good in any case, I think structuring this as a Kubernetes `Job` as the sort of defacto way to run this may have some merit.

Reply:
Agree that that's a natural extension point, but I think even a k8s job would be based around a CLI.